WooCommerce/Classes/ViewRelated/Themes/ThemesPreviewView.swift

```diff
@@ -126,6 +140,61 @@ struct ThemesPreviewView: View {
             }
             .navigationBarTitleDisplayMode(.inline)
         }
+        .task { await viewModel.fetchPages() }
+        .sheet(isPresented: $showPagesMenu) {
+            if #available(iOS 16, *) {
+                pagesListSheet
+                    .presentationDetents([.medium, .large])
+            } else {
+                pagesListSheet
+            }
+        }
+    }
+
+    @ViewBuilder
+    private var pageSelector: some View {
+
+        // Here we show only a "Preview" label with no selector, both for loading and error cases.
+        // In the case of page loading error, the home page is still usable, so showing "Preview" is better than nothing.
+        switch viewModel.state {
+        case .pagesLoading, .pagesLoadingError:
+            Text(Localization.preview)
+
+        case .pagesContent:
+            Button(action: { showPagesMenu = true }) {
+                HStack {
+                    Text(viewModel.selectedPage.title).bodyStyle()
+                    Image(uiImage: .chevronDownImage)
+                        .fixedSize()
+                        .bodyStyle()
+                }
+            }
+        }
+    }
+
+    private var pagesListSheet: some View {
+        GeometryReader { geometry in
+            ScrollView {
+                VStack {
+                    Text(Localization.pagesSheetHeading)
+                        .subheadlineStyle()
+                        .padding(Layout.pagesSheetPadding)
+                    ForEach(viewModel.pages) { page in
+                        Button(action: {
+                            viewModel.setSelectedPage(page: page)
+                            showPagesMenu = false
+                        }, label: {
+                            Text(page.title)
+                                .bodyStyle()
+                                .frame(maxWidth: .infinity, alignment: .leading)
+                        })
+                        .padding(Layout.contentPadding)
+                    }
+                    Spacer()
+                }
+                .frame(minHeight: geometry.size.height)
```

Comment:
I am curious. Do we need to set the minHeight using `GeometryReader`? What happens if we don't set it?

Reply:
The `.presentationDetents` shows the sheet at a predefined height (medium and large).

When the Pages list is very long, at medium sheet height, the list is somehow shown vertically centered and the top label are cut-off. This is a solution for that.

Video of the issue that this commit solves:


https://github.com/woocommerce/woocommerce-ios/assets/266376/7466d813-1664-43fa-831e-f8658813ddd5

Reply:
Thanks for the explanation, Hafiz! 

I removed the GeometryReader, and I only kept the ScrollView

```swift
    private var pagesListSheet: some View {
        ScrollView {
            VStack {
                Text(Localization.pagesSheetHeading)
                    .subheadlineStyle()
                    .padding(Layout.pagesSheetPadding)
                ForEach(viewModel.pages) { page in
                    Button(action: {
                        viewModel.setSelectedPage(page: page)
                        showPagesMenu = false
                    }, label: {
                        Text(page.title)
                            .bodyStyle()
                            .frame(maxWidth: .infinity, alignment: .leading)
                    })
                    .padding(Layout.contentPadding)
                }
                Spacer()
            }
        }
    }
```

I don't see this issue in iOS 16.4 simulator. Can you let me know what I am missing here? 

https://github.com/woocommerce/woocommerce-ios/assets/524475/c511d413-6586-44a6-9af4-e6eea5c18798

@itsmeichigo Is using `GeometryReader` required?

Reply:
Conversely, when the Pages list is too short, this stretching issue happened, and it's fixed also by this commit:

https://github.com/woocommerce/woocommerce-ios/assets/266376/4412340b-7305-4da0-a244-a38b0091b950

Reply:
Huh, weird, I wonder why this happened in my Simulator. It is also in 16.4.

Reply:
> Conversely, when the Pages list is too short, this stretching issue happened, and it's fixed also by this commit:

I assume that you mean the code without `ScrollView`. Without the scroll view the `VStack` is trying to occupy all the space which is expected behaviour. 

BTW, my question is about the use of `GeometryReader` and the `.frame(minHeight: geometry.size.height)` modifier.

Reply:
Oh I think I see what you meant. ScrollView alone is enough? Testing.

Reply:
Updated in https://github.com/woocommerce/woocommerce-ios/pull/11492/commits/9075f71c20b4a80147bd700e141159019069a0d1

Reply:
Thank you!